provider/provider.go

```diff
@@ -128,19 +128,33 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 	logging.V(9).Infof("%s executing", label)
 
 	inputs, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{
-		KeepUnknowns: false,
+		KeepUnknowns: true,
 		SkipNulls:    true,
 	})
 	if err != nil {
 		return nil, err
 	}
 
+	if inputs["build"].ContainsUnknowns() {
```

Comment:
I'm surprised that `Check` is sent with unknowns with preview disabled. That feels like an odd choice for the engine.

Reply:
https://github.com/pulumi/pulumi-docker/blob/a5f956833c067a15d9c88fe45641b43b1e35ed10/provider/hybrid.go#L93-#L93 is what we tell the engine we support here.